### ISSUE TITLE
Much Richer Sample Clinical Data

### ISSUE THREAD

**Issue by gwaybio** [cognoma/cancer-data#38]
Stumbled upon [snaptron](https://twitter.com/GreeneScientist/status/832635971883593728) today and eventually found my way to this [resource](http://snaptron.cs.jhu.edu/data/tcga/).

There are many variables curated here measured on each sample  (in `samples.tsv`) including treatment (both specific therapeutic agent, and class of therapy (e.g. chemotherapy, immunotherapy, etc.). I know that @yigalron was very interested in this particular data...

**Comment by dhimmel** [comment]
Wow `http://snaptron.cs.jhu.edu/data/tcga/samples.tsv` does have lot's of variables! Do you know whether this resource will be continually maintained?

**Comment by gwaybio** [comment]
Not sure but perhaps @ChristopherWilks will know? we could figshare it

**Comment by dhimmel** [comment]
@gwaygenomics here are the following attributes I care about:

1. The data is public domain and released under a CC0 license if necessary.
2. The creators are committed to providing support
3. The creators are committed to updating the resource to fix issues and incorporate new TCGA data. Preferably, there is some established workflow to facilitate updates.

**Comment by ChristopherWilks** [comment]
Good to see Snaptron and/or its data could be useful for you folks.
@nellore and @lcolladotor were primarily the ones who put samples.tsv together, originally for the recount project, https://jhubiostatistics.shinyapps.io/recount/

As far as maintaining, the project as a whole will be maintained in terms of the expression and junction data for various data sources (SRA, GTEx, TCGA).  I'll defer to @nellore and @lcolladotor regarding updates to the metadata, specifically TCGA (not surprisingly, this was the hardest metadata to integrate and there was a non-trivial effort put into it).

**Comment by gwaybio** [comment]
thanks for the quick response @ChristopherWilks - yes, I certainly understand that collecting that metadata was no small feat!

I also agree with those 3 points @dhimmel raises. Would be great to incorporate this data to cognoma if possible!

**Comment by lcolladotor** [comment]
Hi,

Sorry for the delay. We had to discuss internally some of your concerns with @jtleek, @nellore and others.

Basically, our answer is no, we won't use a CC0 license due to the lack of attribution. We are providing support to https://jhubiostatistics.shinyapps.io/recount/ according to the needs of that project. So we'll most likely fix issues with the data in `recount2` if they arise and will be glad to take pull requests to fix any issues with the TCGA metadata in particular. If we update the TCGA RNA-seq data in `recount2` at some point, we would update the metadata accordingly. But given the cost of doing so, I assume that it won't happen anytime soon if at all.

So basically, you can use the TCGA metadata we cleaned at your own risk if you cite our work. Currently the manuscript is available as a [pre-print](http://biorxiv.org/content/early/2016/08/08/068478) but it is in press right now and will appear published soon. 

Best,
Leonardo

PS The related GitHub repositories to `recount2` are:

* https://github.com/leekgroup/recount-website
* https://github.com/leekgroup/recount-analyses
* https://github.com/nellore/runs
* https://github.com/leekgroup/recount-contributions

There's also the `recount` Bioconductor package at https://github.com/leekgroup/recount and http://bioconductor.org/packages/recount

**Comment by dhimmel** [comment]
@lcolladotor thanks for the detailed information. The links are really helpful, and it's great to see that all your work is on GitHub.

> Basically, our answer is no, we won't use a CC0 license due to the lack of attribution.

Very understandable. Have you considered releasing your data under a [Creative Commons Attribution](https://creativecommons.org/licenses/by/4.0/) (CC BY) or [Open Data Commons Attribution](https://opendatacommons.org/licenses/by/summary/) (ODC-BY) License? Both of these meet the [Open Definition](http://opendefinition.org/) and allow reuse as long as attribution is provided.

As it stands without a license, it is potentially copyright infringement to reuse your datasets. The situation is complicated, since the data is likely not subject to copyright in United States, but may be subject to copyright in Europe. Additionally, fair use may protect reuse, but that varies by jurisdiction and is a subjective & non-deterministic judgement.

Therefore, an open license can help others know with greater certainty what reuse is permissible. I know most researchers aren't excited to divert precious time to understanding legalities. But I bring it up because with a little effort up front, I think we can save a lot of headaches down the road and help advance the data sciences.

**Comment by lcolladotor** [comment]
Hi,

Thank you for your detailed reply. We have decided to use the CC BY 4.0 license for the data in the `recount2` project. For the code we'll use the MIT license.

In more detail:

* https://jhubiostatistics.shinyapps.io/recount/ now mentions that the data license is CC BY 4.0 using the text from https://raw.githubusercontent.com/cc-archive/legalcode/master/by_4.0.txt.
* https://github.com/leekgroup/recount-website and https://github.com/leekgroup/recount-contributions now have an MIT license added via https://help.github.com/articles/adding-a-license-to-a-repository/.
* https://github.com/leekgroup/recount-analyses which is a mix of code and files, but is intended as a supplementary website for the paper has a CC BY 4.0 license.
* https://github.com/nellore/runs already had an MIT license.
* https://github.com/leekgroup/recount already had an Artistic-2.0 license, as per commonly done in Bioconductor packages.

In relation to the TCGA metadata, the code for creating it is in https://github.com/leekgroup/recount-website. The resulting table is downloadable via https://jhubiostatistics.shinyapps.io/recount/. Specifically via http://duffel.rail.bio/recount/TCGA/TCGA.tsv. An R formatted version can be downloaded with:

```R
## Install if needed
source("https://bioconductor.org/biocLite.R")
biocLite('recount')

library('recount')
metadata <- all_metadata('TCGA')

## Some quick info about it
> dim(metadata)
[1] 11284   864
```

Best,
Leonardo

**Comment by dhimmel** [comment]
@lcolladotor thanks! I know licensing is a hassle. Really appreciate that you took the time to unambiguously and openly license recount2.
